pkg/authenticator/config/config.go

```diff
@@ -68,9 +94,12 @@ func NewFromEnv() (*Config, error) {
 
 	containerMode := os.Getenv("CONTAINER_MODE")
 
-	conjurVersion := os.Getenv("CONJUR_VERSION")
-	if len(conjurVersion) == 0 {
-		conjurVersion = "5"
+	conjurVersion := DefaultConjurVersion
+	// Only '4' and '5' are supported, with empty values using the default of '5'
+	if version := os.Getenv("CONJUR_VERSION"); version == "4" {
+		conjurVersion = version
```

Comment:
I understand that this gives the intended result (the tests make that clear, if nothing else)

but the logic seems odd to me, because it is not clear that when CONJUR_VERSION is set to 5 we also set version = CONJUR_VERSION

I know it is simpler the way you have it written, and achieves the same end result - but I feel it could be clearer what the intended behavior is, even if it ends up being a bit more verbose.

but not a dealbreaker, in any case

Reply:
I agree, the logic would be clearer and more extensible to future versions with a switch, something like this (leave out the comments if it's too verbose):
```
	conjurVersion := os.Getenv("CONJUR_VERSION")
	switch conjurVersion {
	case "4": // Valid
	case "5": // Valid
	case "":  // Use default
		conjurVersion = DefaultConjurVersion
	default: // Unknown version
		return nil, log.RecordedError(log.CAKC021E, fmt.Errorf("invalid conjur version"))
	}
```